packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -138,27 +169,80 @@ export class TableLayoutManager<TData extends TableRecord> {
         let availableSpace = 0;
         if (requestedResizeAmount > 0) {
             // size right
-            return requestedResizeAmount;
+            return this.hasResizableColumnToLeft(this.leftColumnIndex!)
+                ? requestedResizeAmount
+                : 0;
         }
 
         // size left
-        let currentIndex = this.leftColumnIndex!;
-        while (currentIndex >= 0) {
-            const columnInitialWidths = this.initialColumnWidths[currentIndex]!;
-            availableSpace
-                += columnInitialWidths.initialPixelWidth
-                - columnInitialWidths.minPixelWidth;
-            currentIndex -= 1;
+        if (!this.hasResizableColumnToRight(this.rightColumnIndex!)) {
+            return 0;
+        }
+
+        for (let i = this.leftColumnIndex!; i >= 0; i--) {
+            const columnInitialWidths = this.initialColumnWidths[i]!;
+            if (!columnInitialWidths.resizingDisabled) {
+                availableSpace
+                    += columnInitialWidths.initialPixelWidth
+                    - columnInitialWidths.minPixelWidth;
+            }
         }
         return Math.max(requestedResizeAmount, -availableSpace);
     }
 
+    /**
+     * Gets the index of the first resizable column starting with
+     * `columnIndex` and moving to the left. If no resizable column
+     * is found, returns -1.
+     */
+    private getFirstLeftResizableColumnIndex(
+        columnIndex: number,
+        visibleColumns = this.visibleColumns
+    ): number {
+        for (let i = columnIndex; i >= 0; i--) {
+            const column = visibleColumns[i];
+            if (!column) {
```

Comment:
How could `column` be falsy here? I don't think we can index outside the array; is there a case where columns in the array are undefined? Or are we just forced to handle this case because TypeScript can't determine whether it'll always be valid?

Reply:
It really shouldn't be. TypeScript forces us to handle that case, but technically a caller could pass a `columnIndex` that is greater than the number of visible columns.